server/api/utils/clients/async_nuclio.py

```diff
@@ -18,35 +18,100 @@
 import aiohttp
 
 import mlrun.common.schemas
+import mlrun.errors
+import mlrun.runtimes.nuclio.api_gateway
 import mlrun.utils
 from mlrun.utils import logger
 
 NUCLIO_API_SESSIONS_ENDPOINT = "/api/sessions/"
-NUCLIO_API_GATEWAYS_ENDPOINT = "/api/api_gateways/"
+NUCLIO_API_GATEWAYS_ENDPOINT = "/api/api_gateways/{api_gateway}"
 API_GATEWAY_NAMESPACE_HEADER = "X-Nuclio-Api-Gateway-Namespace"
 NUCLIO_PROJECT_NAME_HEADER = "X-Nuclio-Project-Name"
+NUCLIO_PROJECT_NAME_LABEL = "nuclio.io/project-name"
+IGUAZIO_USERNAME_LABEL = "iguazio.com/username"
+MLRUN_CREATED_LABEL = "mlrun-created"
 
 
 class Client:
     def __init__(self, auth_info: mlrun.common.schemas.AuthInfo):
         self._session = None
+        self._username = auth_info.username
         self._auth = aiohttp.BasicAuth(auth_info.username, auth_info.session)
         self._logger = logger.get_child("nuclio-client")
         self._nuclio_dashboard_url = mlrun.mlconf.nuclio_dashboard_url
+        self._nuclio_domain = urllib.parse.urlparse(self._nuclio_dashboard_url).netloc
 
-    async def list_api_gateways(self, project_name=None):
+    async def list_api_gateways(
+        self, project_name=None
+    ) -> dict[str, mlrun.common.schemas.APIGateway]:
```

Comment:
why did you choose a `dict` response? is it for an easier permission query in the endpoint?

Reply:
@TomerShor because the Nuclio api returns the dictionary